### ISSUE TITLE
[FEATURE REQUEST] use similar `restricted` syntax for `moduleLayers`

### ISSUE THREAD

**Issue by GerardPaligot** [jraska/modules-graph-assert#129]
If I want to avoid a relation between 2 modules, I can configure the restricted property like this:

```kotlin
moduleGraphAssert {
    restricted = arrayOf(
        // A feature can't have a deps to another feature excepted if it is a core feature
        ":(?:[a-zA-Z]+):(?:[a-zA-Z]+)-feature -X> :(?!core)(?:[a-zA-Z]+):(?:[a-zA-Z]+)-feature",
        // A data can't have a deps to another data excepted if it is a core data
        ":(?:[a-zA-Z]+):(?:[a-zA-Z]+)-data -X> :(?!core)(?:[a-zA-Z]+):(?:[a-zA-Z]+)-data"
    )
}
```

In this case, my module `:{feature-name}:{feature-name}-feature` can't have a dependency to another feature module (excepted for the core feature module) and it is the same for data modules.

I think it could be interesting to have something similar for the `moduleLayers` because I would like to apply multiple rules and not only one path in my module hierarchy.

A propal can be something like this:

```kotlin
moduleGraphAssert {
    moduleLayers = arrayOf(
        ":(?:[a-zA-Z]+):(?:[a-zA-Z]+)-feature -> :(?!core)(?:[a-zA-Z]+):(?:[a-zA-Z]+)-data",
        ":(?:[a-zA-Z]+):(?:[a-zA-Z]+)-feature -> :core:core-feature",
        ":(?:[a-zA-Z]+):(?:[a-zA-Z]+)-data -> :core:core-data",
    )
}
```

I share with you my module architecture graph to show you my module dependency strategy:

![graph](https://user-images.githubusercontent.com/1913901/139403401-00e90b40-365d-43c3-bf9a-cf9cb48a05d4.png)

What do you think?

**Comment by jraska** [comment]
Hi @GerardPaligot please whenever you have a [moment please check](https://github.com/jraska/modules-graph-assert#debugging) if the current plugin state would work for you. [Example of configuration](https://github.com/jraska/modules-graph-assert/pull/130/files#diff-ad98e9ee5c93c7303f021c868ac7b3b8e387c1864794e97c01547939cabf0052R18).

`allowed` property was added. I'm still waiting for feedback from some folks, but I think I start to have an idea how it could look like.

**Comment by jraska** [comment]
Topic discussed with @marrcellogalhardo. 🙇  
Summary: allowed/restricted looks like a better way to do what is usually intended. Deprecating layers makes sense. Module naame/alis will be added as it might be fairly simple #134

**Comment by jraska** [comment]
Resolved by [release 1.7.0](https://github.com/jraska/modules-graph-assert/releases/tag/1.7.0).

```
plugins {
  id "com.jraska.module.graph.assertion" version "1.7.0"
}
```

`allowed` is now experimental and will be finalized with version 2.0.

Thanks @GerardPaligot for the issue as it triggered the right discussions to move the plugin forward 🎉
